### ISSUE TITLE
Fails to find install plan without guidance

### ISSUE THREAD

**Issue by wereHamster** [haskell/cabal#2736]
```
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: mole-0.0.1 (user goal)
trying: kraken-0.0.1 (dependency of mole-0.0.1)
trying: aeson-0.9.0.1 (dependency of kraken-0.0.1)
trying: vector-0.11.0.0 (dependency of aeson-0.9.0.1)
trying: snap-0.14.0.6 (dependency of mole-0.0.1)
next goal: vector-algorithms (dependency of snap-0.14.0.6)
rejecting: vector-algorithms-0.7, 0.6.0.4, 0.6.0.3, 0.6.0.2, 0.6.0.1, 0.5.4.2
(conflict: vector==0.11.0.0, vector-algorithms => vector>=0.6 && <0.11)
rejecting: vector-algorithms-0.5.4.1, 0.5.4, 0.5.3.1, 0.5.3 (conflict:
vector==0.11.0.0, vector-algorithms => vector>=0.6 && <0.10)
rejecting: vector-algorithms-0.5.2, 0.5.1, 0.5.0, 0.4 (conflict:
vector==0.11.0.0, vector-algorithms => vector>=0.6 && <0.8)
rejecting: vector-algorithms-0.3.4, 0.3.3, 0.3.2, 0.3.1, 0.3 (conflict: snap
=> vector-algorithms>=0.4 && <0.8)
Backjump limit reached (change with --max-backjumps).
```

Even though my package (mole) doesn't directly depend on `vector`, if I add a `vector <0.11` constraint to my cabal file then cabal can come up with a valid install plan. Note that it worked a few days ago, when I started developing the package. This morning I wiped my sandbox to do a clean install of the dependencies and I couldn't anymore.

Here is the cabal file: https://github.com/wereHamster/mole/blob/master/mole.cabal.

**Comment by kosmikus** [comment]
Well, you're reaching the backjump limit, which is somewhat arbitrary. Does `--reorder-goals` or `--max-backjumps=-1` or a combination of the two work?

**Comment by grayjay** [comment]
I tried to reproduce this with the `--index-snapshot` feature from #3378.  I didn't notice any difference in behavior between July 26th, 2015 and several days earlier, except that `kraken` wasn't uploaded until July 25th.  If it was installed first, that might have helped cabal find a solution.  When I added `kraken` as an `add-source` dependency and tried to install `mole`, cabal reached the backjump limit, unless it was given a constraint on `vector`.  However, `--reorder-goals` allowed it to find a solution within ten seconds.
